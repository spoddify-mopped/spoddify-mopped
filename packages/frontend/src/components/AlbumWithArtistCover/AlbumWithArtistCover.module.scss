span.artist {
  &:hover {
    text-decoration: underline;
  }
}
