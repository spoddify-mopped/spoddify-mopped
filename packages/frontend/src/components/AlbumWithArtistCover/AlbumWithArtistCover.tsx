import { Album } from '../../clients/api.types';
import Cover from '../Cover/Cover';
import React from 'react';
import styles from './AlbumWithArtistCover.module.scss';

type Props = {
  item: Album;
  onAlbumClick: (id: string) => void;
  onArtistClick: (id: string) => void;
};

const AlbumWithArtistCover = ({
  item,
  onAlbumClick,
  onArtistClick,
}: Props): React.ReactElement => {
  const renderArtistSubtitle = () => {
    return item.artists.map((artist, index) => (
      <>
        <span
          className={styles.artist}
          onClick={() => onArtistClick(artist.id)}
        >
          {artist.name}
        </span>
        {index !== item.artists.length - 1 ? ', ' : ''}
      </>
    ));
  };

  return (
    <Cover
      item={{
        clickableSubtitle: true,
        id: item.id,
        image: item.imageUrl,
        name: item.name,
        onClick: () => {
          onAlbumClick(item.id);
        },
        subtitle: renderArtistSubtitle(),
      }}
    />
  );
};

export default AlbumWithArtistCover;
