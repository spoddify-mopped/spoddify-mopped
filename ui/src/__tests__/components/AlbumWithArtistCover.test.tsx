import AlbumWithArtistCover from '../../components/AlbumWithArtistCover/AlbumWithArtistCover';
import React from 'react';
import renderer from 'react-test-renderer';

it('AlbumWithArtistCover renders required properties and no artists', () => {
  const tree = renderer
    .create(
      <AlbumWithArtistCover
        item={{
          artists: [],
          id: 'some id',
          name: 'some name',
        }}
        onAlbumClick={() => ''}
        onArtistClick={() => ''}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('AlbumWithArtistCover renders required properties and image', () => {
  const tree = renderer
    .create(
      <AlbumWithArtistCover
        item={{
          artists: [],
          id: 'some id',
          imageUrl: 'some image url',
          name: 'some name',
        }}
        onAlbumClick={() => ''}
        onArtistClick={() => ''}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('AlbumWithArtistCover renders required properties and one artist', () => {
  const tree = renderer
    .create(
      <AlbumWithArtistCover
        item={{
          artists: [
            {
              id: 'some artist id',
              name: 'some artist name',
            },
          ],
          id: 'some id',
          name: 'some name',
        }}
        onAlbumClick={() => ''}
        onArtistClick={() => ''}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('AlbumWithArtistCover renders required properties and multiple artists', () => {
  const tree = renderer
    .create(
      <AlbumWithArtistCover
        item={{
          artists: [
            {
              id: 'some artist id',
              name: 'some artist name',
            },
            {
              id: 'another artist id',
              name: 'another artist name',
            },
          ],
          id: 'some id',
          name: 'some name',
        }}
        onAlbumClick={() => ''}
        onArtistClick={() => ''}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
