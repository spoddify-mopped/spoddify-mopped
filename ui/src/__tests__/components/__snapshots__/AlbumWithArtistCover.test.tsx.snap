// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumWithArtistCover renders required properties and image 1`] = `
<div
  className="cover "
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="some image url"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  />
</div>
`;

exports[`AlbumWithArtistCover renders required properties and multiple artists 1`] = `
<div
  className="cover "
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="cover_replacement.png"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  >
    <span
      className="artist"
      onClick={[Function]}
    >
      some artist name
    </span>
    , 
    <span
      className="artist"
      onClick={[Function]}
    >
      another artist name
    </span>
    
  </span>
</div>
`;

exports[`AlbumWithArtistCover renders required properties and no artists 1`] = `
<div
  className="cover "
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="cover_replacement.png"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  />
</div>
`;

exports[`AlbumWithArtistCover renders required properties and one artist 1`] = `
<div
  className="cover "
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="cover_replacement.png"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  >
    <span
      className="artist"
      onClick={[Function]}
    >
      some artist name
    </span>
    
  </span>
</div>
`;
